HesoyamHospital/GraphicEditor/Service/DrawingShapesService.cs

```diff
@@ -105,6 +115,55 @@ public Shape DrawShapes(GraphicalObject graphicalObject)
 
             }
         }
+
+        public void DoubleClick(object sender, MouseButtonEventArgs e, string roomName)
+        {
+            Rectangle rectangle = sender as System.Windows.Shapes.Rectangle;
+            List<InventoryItemDTO> result = new List<InventoryItemDTO>();
+            if (e.ClickCount == 2 && rectangle.Name.Contains("room"))
+            {
+                RoomService roomService = Backend.AppResources.getInstance().roomService;
+                Room room = roomService.GetRoomByName(roomName);
+                long id = room.Id;
+
+                List<Medicine> medicine = (List<Medicine>)medicineService.GetMedicinesByRoom(id);
+                List<InventoryItem> inventoryItems = (List<InventoryItem>)inventoryService.GetInventoryItemsByRoom(roomName);
+
+                List<InventoryItemDTO> inventoryItemDTO = ConvertToIventoryItemToDTO(inventoryItems);
+                List<InventoryItemDTO> medicineDTO = ConvertToMedicineToDTO(medicine, roomName);
+                result.AddRange(inventoryItemDTO);
+                result.AddRange(medicineDTO);
+
+                OverviewEquipmentAndMedicine window = new OverviewEquipmentAndMedicine();
+                window.dataGridOverview.ItemsSource = result;
+                window.Show();
+
+            }
+        }
+
+        public List<InventoryItemDTO> ConvertToIventoryItemToDTO(List<InventoryItem> inventoryItems) {
```

Comment:
I suppose method name should be Convert**From**IventoryItemToDTO. Same applies to the method below.
These kind of methods should be probably placed in a separate mapper class which will have a responsibility of converting from model to DTO and vice versa.
Take a look at Web application project. They have implemented something similiar.

Also,  make sure to set methods to be private if they are not supposed to be called from anywhere else and are just a helper method.